test/e2e/reboots/kernel_parameter_remove.go

```diff
@@ -0,0 +1,157 @@
+			)
+
+			ginkgo.By("getting a list of worker nodes")
+			nodes, err := util.GetNodesByRole(cs, "worker")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(len(nodes)).NotTo(gomega.BeZero(), "number of worker nodes is 0")
+
+			workerMachinesOrig, err := util.GetUpdatedMachineCountForPool(cs, "worker")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			node = &nodes[0]
+			ginkgo.By(fmt.Sprintf("getting a tuned pod running on node %s", node.Name))
+			pod, err := util.GetTunedForNode(cs, node)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By(fmt.Sprintf("getting the current %s value in pod %s", procCmdline, pod.Name))
+			cmdlineOrig, err := util.GetFileInPod(pod, procCmdline)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			util.Logf(fmt.Sprintf("%s has %s: %s", pod.Name, procCmdline, cmdlineOrig))
+
+			ginkgo.By(fmt.Sprintf("labelling node %s with label %s", node.Name, nodeLabelRealtime))
+			_, _, err = util.ExecAndLogCommand("oc", "label", "node", "--overwrite", node.Name, nodeLabelRealtime+"=")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By(fmt.Sprintf("creating custom parent profile %s", profileParent))
+			_, _, err = util.ExecAndLogCommand("oc", "create", "-n", ntoconfig.OperatorNamespace(), "-f", profileParent)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By(fmt.Sprintf("creating custom MachineConfigPool %s", mcpRealtime))
+			_, _, err = util.ExecAndLogCommand("oc", "create", "-f", mcpRealtime)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("waiting for worker-rt MachineConfigPool UpdatedMachineCount == 1")
+			err = util.WaitForPoolUpdatedMachineCount(cs, "worker-rt", 1)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By(fmt.Sprintf("getting the current %s value in pod %s", procCmdline, pod.Name))
+			cmdlineNew, err := util.GetFileInPod(pod, procCmdline)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			util.Logf("%s has %s: %s", pod.Name, procCmdline, cmdlineNew)
+
+			// Check the key kernel parameters for the parent profile to be present in /proc/cmdline
+			ginkgo.By("ensuring the custom worker parent profile was set")
+			for _, v := range []string{paramKeep, paramRemove} {
+				gomega.Expect(strings.Contains(cmdlineNew, v)).To(gomega.BeTrue(), "missing '%s' in %s: %s", v, procCmdline, cmdlineNew)
+			}
+
+			ginkgo.By(fmt.Sprintf("creating custom child profile %s", profileChild))
+			_, _, err = util.ExecAndLogCommand("oc", "create", "-n", ntoconfig.OperatorNamespace(), "-f", profileChild)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("ensuring the custom worker child profile was set")
+			err = wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+				cmdlineNew, err = util.GetFileInPod(pod, procCmdline)
+				if err != nil {
+					explain = err.Error()
+					return false, nil
+				}
+				if strings.Contains(cmdlineNew, paramRemove) {
```

Comment:
In all the contains check, do we need to consider also the order of the parameters ?  There are some use cases i believe like hugepages where the order is important.
If so then I suggest to use regular expressions instead of a string.contains check

Reply:
That's a great point, thank you.  Fixed.